pkg/llms/dashscope/api.go

```diff
@@ -62,9 +68,105 @@ func (z *DashScope) Call(data []byte) (llms.Response, error) {
 	if err := params.Unmarshal(data); err != nil {
 		return nil, err
 	}
-	return do(context.TODO(), DashScopeChatURL, z.apiKey, data, z.sse)
+	return do(context.TODO(), DashScopeChatURL, z.apiKey, data, z.sse, false)
 }
 
 func (z *DashScope) Validate() (llms.Response, error) {
 	return nil, errors.New("not implemented")
 }
+
+func (z *DashScope) CreateEmbedding(ctx context.Context, inputTexts []string, query bool) ([]Embeddings, error) {
+	textType := TextTypeDocument
+	if query {
+		textType = TextTypeQuery
+	}
+	reqBody := EmbeddingRequest{
+		Model: EmbeddingV1,
+		Input: EmbeddingInput{
+			EmbeddingInputSync: &EmbeddingInputSync{
+				Texts: inputTexts,
+			},
+		},
+		Parameters: EmbeddingParameters{
+			TextType: textType,
+		},
+	}
+	data, err := json.Marshal(reqBody)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := req(ctx, DashScopeTextEmbeddingURL, z.apiKey, data, false, false)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	respData := &EmbeddingResponse{}
+	if err := json.NewDecoder(resp.Body).Decode(respData); err != nil {
+		return nil, err
+	}
+	if respData.StatusCode != 200 && respData.StatusCode != 0 {
+		return nil, errors.New(respData.Message)
+	}
+	return respData.Output.Embeddings, nil
+}
+
+func (z *DashScope) CreateEmbeddingAsync(ctx context.Context, inputURL string, query bool) (taskID string, err error) {
+	textType := TextTypeDocument
+	if query {
+		textType = TextTypeQuery
+	}
+	reqBody := EmbeddingRequest{
```

Comment:
The async mode only allows url as the input.Right?

Reply:
yes, async -> url / sync-> text.